tools/chplcheck/src/rule_types.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright 2024 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import typing
+from dataclasses import dataclass
+
+import chapel
+from fixits import ChapelFixit
+
+
+@dataclass
+class FullBasicRuleResult:
+    check: bool
+    fixit: typing.Optional[
+        typing.Union[ChapelFixit, typing.List[ChapelFixit]]
+    ] = None
+
+
+BasicRuleResult = typing.Union[bool, ChapelFixit]
```

Comment:
I've noted elsewhere to rename other things to this. I didn't spot where `BasicRuleResult` is used without the `Full` prefix. Either way I suspect it's an internal alias, which means you could name it something less nice to make room for user code that needs to construct what is today called the `FullBasicRuleResult`.